lib/puppet/environments.rb

```diff
@@ -435,23 +447,37 @@ def get_conf(name)
     # Creates a suitable cache entry given the time to live for one environment
     #
     def entry(env)
-      ttl = (conf = get_conf(env.name)) ? conf.environment_timeout : Puppet.settings.value(:environment_timeout)
+      mru_entry = false
```

Comment:
Also it might be a bit shorter to initialize `mru_entry` to the result of `Puppet.settings.set_by_config?(:environment_ttl)` which we use below. Then `ttl` becomes:

```ruby
ttl = if mru_entry
        Puppet[:environment_ttl]
      elsif (conf = get_conf(env.name))
        conf.environment_timeout
      else
        Puppet[:environment_timeout]
      end
```
Or have a `mru_entry?` method which returns `Puppet.settings.set_by_config?(:environment_ttl)`. I don't have a specific preference.

Reply:
Good suggestion, looks better now